Fix terminal_width to work properly on windows

win93 opened the pull request:
- Windows shell uses NUL instead of /dev/null to discard output. Before, this method would print "The system cannot find the file specified," which defeats the purpose of the redirection
- Running `stty size` on Windows takes a non-trivial amount of time, so for Ruby >= 1.9.3, IO.console is used instead on all platforms
- As with #1, both CMD and powershell are also horribly broken, and subtracting 1 from the columns is required to keep the bar from wrapping

win93 committed:
Fix terminal_width to work properly on windows

- Windows shell uses NUL instead of /dev/null to discard output. Before, this method would print "The system cannot find the file specified," which defeats the purpose of the redirection
- Running `stty size` on Windows takes a non-trivial amount of time, so for Ruby >= 1.9.3, IO.console is used instead on all platforms
- As with #1, both CMD and powershell are also horribly broken, and subtracting 1 from the columns is required to keep the bar from wrapping

m-o-e commented on lib/powerbar.rb:
```diff
@@ -308,14 +308,20 @@ def h_done
   def terminal_width
+    require 'io/console'
+    rows, cols = IO.console.winsize
+    cols - 1
```
Are you sure `cols - 1` is needed on _all_ platforms? 🤔

win93 replied:
No I am not sure, but #1 says its needed for OS X as well. I can check on a linux machine in a sec.

win93 replied:
It's not necessary on linux, at least through PuTTY. :/ Not sure what to recommend.

m-o-e replied:
I just tested it on OSX (iTerm and Terminal.app), and it's not needed here either.

I would suggest changing the line to cols -= 1 if Gem.win_platform?

Can you test if this works correctly on win and linux? (it does on OSX here)

win93 replied:
Actually, it doesn't seem to be necessary on OS X either, at least with up-to-date versions of Terminal.app and iTerm2.app

m-o-e commented on lib/powerbar.rb:
```diff
@@ -308,14 +308,20 @@ def h_done
+    require 'io/console'
+    rows, cols = IO.console.winsize
+    cols - 1
+  rescue LoadError #io/console not in ruby < 1.9.3
```
I think we should just drop support for Ruby <1.9.3 and for Solaris at this point.

If someone needs this gem on these old platforms they can just stick to an older gem version.

win93 replied:
That simplifies the method a bit. I'll push a commit that has terminal width using just io/console.

win93 replied:
Actually, it doesn't seem to be necessary on OS X either, at least with up-to-date versions of Terminal.app and iTerm2.app

win93 committed:
Drop support for ruby < 1.9.3 at request of package maintainer

m-o-e commented:
I just tested it on OSX (iTerm and Terminal.app), and it's not needed here either.

I would suggest changing the line to `cols -= 1 if Gem.win_platform?`

Can you test if this works correctly on win and linux? (it does on OSX here)

win93 committed:
Restrict column width off by 1 to just windows

Situations I have tested that this is not required:
- MacOS Terminal.app
- MacOS iTerm2.app
- Windows PuTTY to a linux machine
- local linux tty

Situations where it is required:
- Windows Command Prompt
- Windows Powershell

m-o-e commented:
Nice!   Looks good to me now, thank you very much for that contribution!

m-o-e commented:
Merged and cut a new release. Major version bumped since we're dropping support for Solaris and old rubies.

m-o-e commented:
Ah, thanks, good catch indeed! 👍 

Fixed and released again.

m-o-e commented:
And deleted v2.0.0 from rubygems
